src/app/services/baw-api/security/security.service.ts

```diff
@@ -58,7 +63,58 @@ export class SecurityService extends BawApiService<SessionUser> {
    * @param details Details provided by login form
    */
   public signIn(details: LoginDetails): Observable<void> {
-    return this.handleAuth(this.apiCreate(signInEndpoint(), details));
+    function setFormData(page: any, body: URLSearchParams) {
+      const err: ApiErrorDetails = {
+        status: apiReturnCodes.unknown,
+        message: "Unable to retrieve authenticity token for login request",
+      };
+
+      // Catch wrong type of response
+      if (typeof page !== "string") {
+        throwError(err);
+      }
+
+      // Extract auth token if exists
+      const token = page.match(/name="authenticity_token" value="(.+?)"/);
+      if (typeof token[1] !== "string") {
```

Comment:
a truthy check is probably sufficent here? or a check for null/undefined? regex match will always return a string